packages/react/src/components/Suggestion/Suggestion.tsx

```diff
@@ -0,0 +1,100 @@
+import { getDatalistValue, syncDatalistState } from '@u-elements/u-datalist';
+import cl from 'clsx/lite';
+import {
+  createContext,
+  forwardRef,
+  useCallback,
+  useId,
+  useRef,
+  useState,
+} from 'react';
+import type { DefaultProps } from '../../types';
+
+type SuggestionContextType = {
+  singular?: string;
+  plural?: string;
+  listId?: string;
+  inputRef?: React.RefObject<HTMLInputElement | null>;
+  setListId?: (id: string) => void;
+  handleFilter?: (input?: HTMLInputElement | null) => void;
+};
+
+export const SuggestionContext = createContext<SuggestionContextType>({});
+
+export type SuggestionProps = DefaultProps &
+  React.HTMLAttributes<HTMLDivElement> & {
+    /**
+     * The screen reader announcement for singular suggestion, where %d is the number of suggestions
+     * @default '%d forslag'
+     */
+    singular?: string;
+    /**
+     * The screen reader announcement for plural suggestions, where %d is the number of suggestions
+     * @default '%d forslag'
+     */
+    plural?: string;
+    /**
+     * Filter options, either true | false or custom callback () => boolean
+     * @default true
+     */
+    filter?:
+      | boolean
+      | ((args: {
+          index: number;
+          text: string;
+          value: string;
+          optionElement: HTMLOptionElement;
+          input: HTMLInputElement;
+        }) => boolean);
+  };
+
+export const Suggestion = forwardRef<HTMLDivElement, SuggestionProps>(
+  function Suggestion(
+    {
+      className,
+      singular = '%d forslag',
+      plural = '%d forslag',
+      filter = true,
+      ...rest
+    },
+    ref,
+  ) {
+    const [listId, setListId] = useState(useId());
+    const inputRef = useRef<HTMLInputElement | null>(null);
+
+    const handleFilter = useCallback(
+      (input?: HTMLInputElement | null) => {
+        const list = input?.list;
+
+        // Let <datalist> handle filtering if filter is true
+        if (filter === true || !list) return;
+
+        // Handle custom filter
+        if (filter !== false)
```

Comment:
Should we use `typeof` here to check if it is a function?

Reply:
It is faster to check against a boolean, and as only `true` | `false` | `() => boolean` is supported, I opted for this solution, but again - you can decide :)